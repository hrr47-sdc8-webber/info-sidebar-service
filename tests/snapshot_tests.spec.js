import React from 'react';
import renderer from 'react-test-renderer';
import MockDate from 'mockdate'


import Website from '../src/js/components/website';
import Time from '../src/js/components/time';
import Address from '../src/js/components/address';
import Directions from '../src/js/components/directions';
import Ratings from '../src/js/components/ratings';
import RestaurantInfo from '../src/js/components/restaurantInfo';
import Map from '../src/js/components/map';
import Phone from '../src/js/components/phone';
import Schedule from '../src/js/components/schedule';

it('renders website component correctly', () => {
  const tree = renderer
    .create(<Website website="http://www.facebook.com" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders time component correctly', () => {
  MockDate.set(1466424490000);
  const tree = renderer
    .create(<Time opening="9:30" closing="11:30" id={1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  MockDate.reset();
});

it('renders address component correctly', () => {
  const tree = renderer
    .create(<Address
      restaurantName="Chucks"
      streetAddress="319 Happy St"
      city="Solon"
      state="Ohio"
      zipCode={44139}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders directions component correctly', () => {
  const tree = renderer
    .create(<Directions streetAddress="319 Happy St" city="Solon" state="Ohio" zipCode={44139} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ratings component correctly', () => {
  const tree = renderer
    .create(<Ratings foodRating="3.5" serviceRating="3.5" decorRating="3.5" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders restaurant information component correctly', () => {
  const tree = renderer
    .create(<RestaurantInfo
      foodRating="3.5"
      serviceRating="3.5"
      decorRating="3.5"
      restaurantName="Happy's"
      writtenReview="Some great food"
      sentence="Some great food"
      neighborhood="Boston"
      typeOfFood="BBQ"
      averagePrice="30"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders map component correctly', () => {
  const tree = renderer
    .create(<Map streetAddress="319 Happy St" city="Solon" state="Ohio" zipCode={44139} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders phone component correctly', () => {
  const tree = renderer
    .create(<Phone telephoneNumber="440-349-0105" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the schedule component correctly', () => {
  MockDate.set('1/30/2000');
  const tree = renderer
    .create(<Schedule
      clicked
      open="9:30"
      close="10:30"
      dayOfTheWeek={0}
      id={Math.random()}
       />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  MockDate.reset();
})
