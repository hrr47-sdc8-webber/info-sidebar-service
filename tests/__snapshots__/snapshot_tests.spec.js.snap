// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders address component correctly 1`] = `
<div
  className="items"
>
  <div />
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-map-marker-alt fa-w-12 icons"
    data-icon="map-marker-alt"
    data-prefix="fas"
    focusable="false"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <a
    className="address__Theme-sc-12r5eww-0 dEGZvt"
    onClick={[Function]}
  >
      319 Happy St, Solon, Ohio, 44139, USA
  </a>
</div>
`;

exports[`renders directions component correctly 1`] = `
<a
  className="directions__Theme-sc-4nmpw5-0 jfWIqy"
  href="https://www.google.com/maps/dir/?api=1&destination=319+Happy+St,Solon,Ohio,44139"
  rel="noreferrer"
  target="_blank"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-directions fa-w-16 icons"
    data-icon="directions"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M502.61 233.32L278.68 9.39c-12.52-12.52-32.83-12.52-45.36 0L9.39 233.32c-12.52 12.53-12.52 32.83 0 45.36l223.93 223.93c12.52 12.53 32.83 12.53 45.36 0l223.93-223.93c12.52-12.53 12.52-32.83 0-45.36zm-100.98 12.56l-84.21 77.73c-5.12 4.73-13.43 1.1-13.43-5.88V264h-96v64c0 4.42-3.58 8-8 8h-32c-4.42 0-8-3.58-8-8v-80c0-17.67 14.33-32 32-32h112v-53.73c0-6.97 8.3-10.61 13.43-5.88l84.21 77.73c3.43 3.17 3.43 8.59 0 11.76z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  Get Directions
</a>
`;

exports[`renders map component correctly 1`] = `
<iframe
  allowFullScreen={true}
  frameBorder="0"
  height="175%"
  src="https://www.google.com/maps/embed/v1/place?key=AIzaSyDPIUHkZ4FNCluRm_Vbm-Hp3pjqg42-Anw
      &q=319+Happy+St,Solon,Ohio,44139"
  style={
    Object {
      "border": "0",
    }
  }
  title="restaurantMap"
  width="100%"
/>
`;

exports[`renders phone component correctly 1`] = `
<a
  className="phone__Theme-sc-168fnlg-0 ccNKSf"
  href="\\"tel:440-349-0105\\""
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-phone fa-w-16 icons"
    data-icon="phone"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  440-349-0105
</a>
`;

exports[`renders ratings component correctly 1`] = `
Array [
  <section
    className="ratings__ReviewBox-sc-1g6gxaz-0 gsLzLV"
  >
    <div
      className="innerBox"
    >
      <div
        className="rating"
      >
        3.5
      </div>
      <div
        className="category"
      >
        Food
      </div>
    </div>
    <div
      className="border-middle"
    />
    <div
      className="innerBox"
    >
      <div
        className="rating"
      >
        3.5
      </div>
      <div
        className="category"
      >
        Decor
      </div>
    </div>
    <div
      className="border-middle"
    />
    <div
      className="innerBox"
    >
      <div
        className="rating"
      >
        3.5
      </div>
      <div
        className="category"
      >
        Service
      </div>
    </div>
  </section>,
  <section
    className="ratings__BorderBottom-sc-1g6gxaz-1 hxwUVe"
  />,
]
`;

exports[`renders restaurant information component correctly 1`] = `
<section
  className="restaurantInfo__MainWrapper-sc-5o8o3k-0 cVdMvm"
>
  <div
    className="name"
  >
    Happy's
  </div>
  <div
    className="describes-restaurant"
  >
    Some great food
  </div>
  <div
    className="horizontal-wrapper"
  >
    <a
      className="categories"
      href="#"
    >
      BBQ
    </a>
    <div
      className="dots"
    >
      •
    </div>
    <a
      className="categories"
      href="#"
    >
      Boston
    </a>
    <div
      className="dots"
    >
      •
    </div>
    <a
      className="categories"
      href="#"
    >
      $
    </a>
  </div>
  <div
    className="horizontal-rule"
  />
  <div
    className="horizontal-rule"
  />
  <img
    alt="The Zagat Logo"
    className="zagat-logo"
    src="/imgs/z-logo-icon-red.svg"
  />
  <div
    className="review-text"
  >
    THE ZAGAT REVIEW
  </div>
  <section
    className="ratings__ReviewBox-sc-1g6gxaz-0 gsLzLV"
  >
    <div
      className="innerBox"
    >
      <div
        className="rating"
      >
        3.5
      </div>
      <div
        className="category"
      >
        Food
      </div>
    </div>
    <div
      className="border-middle"
    />
    <div
      className="innerBox"
    >
      <div
        className="rating"
      >
        3.5
      </div>
      <div
        className="category"
      >
        Decor
      </div>
    </div>
    <div
      className="border-middle"
    />
    <div
      className="innerBox"
    >
      <div
        className="rating"
      >
        3.5
      </div>
      <div
        className="category"
      >
        Service
      </div>
    </div>
  </section>
  <section
    className="ratings__BorderBottom-sc-1g6gxaz-1 hxwUVe"
  />
  <div
    className="written-review"
  >
    Some great food
  </div>
</section>
`;

exports[`renders the schedule component correctly 1`] = `
<section
  className="schedule__Wrapper-sc-11cppfk-0 lhwFWA"
>
  <b>
    <section
      className="schedule__InnerWrapper-sc-11cppfk-1 iFpJci"
    >
      <div>
        Sunday
      </div>
      <section
        className="schedule__DotsWrapper-sc-11cppfk-2 eHdDOo"
      >
        .......................................................................................
      </section>
      <div>
        9:30
        : 
        10:30
      </div>
    </section>
  </b>
  <section
    className="schedule__InnerWrapper-sc-11cppfk-1 iFpJci"
  >
    <div>
      Monday
    </div>
    <section
      className="schedule__DotsWrapper-sc-11cppfk-2 eHdDOo"
    >
      .......................................................................................
    </section>
    <div>
      9:30
      : 
      10:30
    </div>
  </section>
  <section
    className="schedule__InnerWrapper-sc-11cppfk-1 iFpJci"
  >
    <div>
      Tuesday
    </div>
    <section
      className="schedule__DotsWrapper-sc-11cppfk-2 eHdDOo"
    >
      .......................................................................................
    </section>
    <div>
      9:30
      : 
      10:30
    </div>
  </section>
  <section
    className="schedule__InnerWrapper-sc-11cppfk-1 iFpJci"
  >
    <div>
      Wednesday
    </div>
    <section
      className="schedule__DotsWrapper-sc-11cppfk-2 eHdDOo"
    >
      .......................................................................................
    </section>
    <div>
      9:30
      : 
      10:30
    </div>
  </section>
  <section
    className="schedule__InnerWrapper-sc-11cppfk-1 iFpJci"
  >
    <div>
      Thursday
    </div>
    <section
      className="schedule__DotsWrapper-sc-11cppfk-2 eHdDOo"
    >
      .......................................................................................
    </section>
    <div>
      9:30
      : 
      10:30
    </div>
  </section>
  <section
    className="schedule__InnerWrapper-sc-11cppfk-1 iFpJci"
  >
    <div>
      Friday
    </div>
    <section
      className="schedule__DotsWrapper-sc-11cppfk-2 eHdDOo"
    >
      .......................................................................................
    </section>
    <div>
      9:30
      : 
      10:30
    </div>
  </section>
  <section
    className="schedule__InnerWrapper-sc-11cppfk-1 iFpJci"
  >
    <div>
      Saturday
    </div>
    <section
      className="schedule__DotsWrapper-sc-11cppfk-2 eHdDOo"
    >
      .......................................................................................
    </section>
    <div>
      9:30
      : 
      10:30
    </div>
  </section>
</section>
`;

exports[`renders time component correctly 1`] = `
Array [
  <div
    className="items"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-clock fa-w-16 icons"
      data-icon="clock"
      data-prefix="fas"
      focusable="false"
      id="icon-testing"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <h1
      className="status"
      onClick={[Function]}
    >
      Closed Now • 
      9:30 AM
       - 
      11:30 AM
          
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-caret-down fa-w-10 arrow"
      data-icon="caret-down"
      data-prefix="fas"
      focusable="false"
      onAnimationEnd={[Function]}
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>,
  <section
    className="schedule__Wrapper-sc-11cppfk-0 jtSTiZ"
  >
    <div />
  </section>,
]
`;

exports[`renders website component correctly 1`] = `
<a
  className="website__Theme-sc-1xz0c8r-0 fmmanX"
  href="http://www.facebook.com"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-columns fa-w-16 icons"
    data-icon="columns"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M464 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V80c0-26.51-21.49-48-48-48zM224 416H64V160h160v256zm224 0H288V160h160v256z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  http://www.facebook.com
</a>
`;
